### ISSUE TITLE
Ability to specify a planId when creating an EXTERNAL_CHARGE

### ISSUE THREAD

**Issue by sbrossie** [killbill/killbill#1532]
Invoice items (charges) in Kill Bill can either be of type `EXTERNAL_CHARGE` and have no relation with any of the Catalog `Plan`, or be `FIXED`, `RECURRING`, `USAGE` and be tied to a Catalog `Plan` though the `Subscription`. There are situations where it would be really handy to reference a `Plan` without any underlying subscription. One such example would be to have a way to invoice for fixed amounts without the need to create subscriptions.

The call to create an `EXTERNAL_CHARGE` would take a `planId` (`planName`), and it would not specify any `amount`. If the `Plan` has a single phase with a fixed price, then the fixed amount would be used for the `EXTERNAL_CHARGE`.

**Comment by sbrossie** [comment]
Another use case is one where we would like to implement a credit based system, where consumers can buy credits of a certain type (in-advance) and use these credits in exchange of a specific service. There are many such models out there (bridge tolls, prepaid phone, ...).

In such a scenario, we would define specific `Plans` in the catalog to keep the mapping between `price` and # of units bought.

1. Initially, and also as the credits get depleted, the customer would buy credits, which would translate into creating an `EXTERNAL_CHARGE` and associate a `Plan`, which defines the amount of units bought and its associated price.
2. The management of the credits would be left outside of Kill Bill core - e.g a plugin being notified of a the successful payment associated with the `EXTERNAL_CHARGE` and extracting from the `Plan` how many units of a certain types were purchased. The plugin would typically export additional apis to consume credits, and implement the entitlement logic to warn users when credits get low, and/or to cancel the service when exhausted.

**Comment by shaun-forgie** [comment]
@sbrossie so if i understand correctly account credits could be recorded against a given quantity of a specific type of usage [unit type] with the usage unit types and pricng tiers defined in a catalog plan definition. 

For example if a calling plan defined two consumption based usage units:

1. national minutes - with all tier pricing
2. international minutes - with top tier pricing

You could generate an external charge invoice that referenced the plan id and usage unit type.  If the invoice got processed successfully then corresponding account credits for national | international minutes gets automatically created for the account.

So in summary account credits would be extended to manage a dollar amount and given quantities of usage unit types. These would be deducted off invoices as part of the normal invoicing cycles thereby reducing the invoice balance.
